segment.go

```diff
@@ -168,3 +168,19 @@ type DocVisitState interface {
 type StatsReporter interface {
 	ReportBytesWritten(bytesWritten uint64)
 }
+
+type FieldStatsReporter interface {
+	UpdateFieldStats(FieldStats)
+}
+
+type FieldStats interface {
+	Store(statName, fieldName string, value uint64)
```

Comment:
Would Set() be a better name here?

Reply:
Store reflects the nature better because we would only do it once per combination per segment.